### PR TITLE
Improve MMR OCR fallback for residual FN after #213

### LINKED ISSUE #212: [Investigation] Improve remaining MMR FN OCR candidate extraction after #211
## 背景

Follow-up from #208 / PR #211.

PR #211 fixed part of the post-#200 MMR OCR residuals by aggregating configured OCR preprocessing variant candidates instead of accepting the first valid variant.

Post-#211 MMR-only aggregate recorded in #208:

```text
Total Expected: 182
Total Detected: 178
Matched (TP): 171
Missed (FN): 5
Skip Mismatch: 6
Unexpected (FP): 1
Precision: 0.9607
Recall: 0.9396
F1: 0.9500
```

#208 classified the remaining FN cases as OCR candidate missing / crop-preprocessing follow-up rather than CNN gate failures.

## 目的

Post-#211 に残る MMR FN について、OCR candidate が生成されない原因を trace し、小さく一般化できる crop / preprocessing 修正候補を検証する。

## 対象 residual 候補

#208 時点で残 FN とされた候補:

```text
page_001 [0,2,2] expected=4
page_002 [1,5,2] expected=3
page_004 [3,2,2] expected=3
page_009 [8,0,0] expected=3
page_042 [41,8,0] expected=3
```

まず post-#211 の fresh eval で exact residual table を再取得し、上記 5 件がまだ FN として残っているか確認する。

### In

- post-#211 exact residual reconstruction
- FN 5 件の targeted trace
- crop / masked crop / preprocessed variant image / OCR raw output の確認
- OCR candidate missing の原因分類
- 小さく一般化できる crop/preprocessing 修正の試行
- mocked regression または small fixture regression
- MMR-only eval による aggregate 比較

### Out

- #201 / PR #209 manual correction workflow
- #197 divisi/system grouping redesign
- CNN retraining
- full pipeline redesign
- 大量 generated artifacts の commit
- `page_033` FP guard; this is tracked separately

## 初期分類軸

各 FN を以下に分類する。

```text
A. 数字が crop に入っていない
B. crop には入っているが hbar mask / preprocessing で潰れる
C. OCR raw output はあるが candidate selection で落ちる
D. system/staff/measure bbox がズレている
E. GT / accounting 側の確認が必要
```

## Done 条件

- post-#211 exact residual table が記録されている。
- 残 FN について targeted trace が記録されている。
- 各 FN の likely owner が分類されている。
- 小さく安全な修正がある場合は、1 種類の修正に限定して PR 化する。
- 修正する場合は MMR-only aggregate で次を確認する。

```text
FN < 5
TP > 171
FP <= 1
mismatch <= 6
total_detected が不自然に増えない
```

## Notes

- #197 の `page_021` grouping/layout は扱わない。
- #201 manual correction workflow は扱わない。
- `page_033` FP は別 issue で追跡する。
- 大きな画像・ログ・zip は commit せず、issue comment の evidence として扱う。

### PR DESCRIPTION
**Pull request by M763468**
## Summary

This PR implements the bounded production fixes validated in #212 after the #213 one-bar veto merge.

Changes:

- Add current-crop unmasked OCR fallback for MMR numbering when masked OCR variants produce no numeric candidate.
- Add score-gated left-wide unmasked OCR fallback, also only after the prior fallback paths fail.
- Add targeted tests for fallback recovery, fallback non-use when existing masked OCR succeeds, negative-score rejection, and #213 one-bar veto preservation.

This PR intentionally does not attempt to solve the remaining three residual FN with broader component/OCR heuristics. Those have been split to #221.

## Validation

Targeted tests:

```text
PYTHONPATH=. pytest tests/test_issue212_mmr_unmasked_fallback.py tests/test_issue213_mmr_one_bar_veto.py
11 passed
```

Lint:

```text
make lint
All checks passed!
370 files already formatted
```

Full 68-page MMR eval using `pdfscore_pipeline_pytest_dev` and `tools/issue94/eval_all_mmr.py`:

```text
Total Pages:         68
Total Base Measures: 3325
Total Expected:      182
Total Detected:      179
Matched (TP):        173
Missed (FN):         3
Skip Mismatch:       6
Unexpected (FP):     0
Precision:           0.9664804469
Recall:              0.9505494505
F1-Score:            0.9584487535
```

Comparison against post-#213 baseline:

```text
Post-#213 baseline: TP=171, FN=5, mismatch=6, FP=0, F1=0.9526
Candidate:          TP=173, FN=3, mismatch=6, FP=0, F1=0.9584
```

Recovered residual FN:

```text
page_042 [41,8,0] via current-crop unmasked fallback
page_002 [1,5,2] via score-gated left-wide unmasked fallback
```

No new missed keys, no new mismatch keys, no unexpected detections.

#213 one-bar veto remains preserved:

```text
[VETO] P33 S0 M1: Prob=0.54 -> OCR=11 (score=-27.4, one_bar_evidence=2)
```

## Residual / follow-up

Remaining FN are intentionally left for #221:

```text
page_001 [0,2,2] expected_num=4
page_004 [3,2,2] expected_num=3
page_009 [8,0,0] expected_num=3
```

Reason:

- `page_001`: expected `4` did not appear in crop/component diagnostics; wrong candidates such as `2`, `3`, `9`, `13`, `86`, `131`, `191` appeared.
- `page_004`: no expected `3` appeared; OCR stayed empty or non-numeric.
- `page_009`: `3` appeared once in a diagnostic line-removal variant, but with score `-53.65` and as a visually unsafe wide-crop artifact; expanded/left-wide variants also repeatedly read `31`.

## Risk

The new fallbacks are deliberately ordered after the existing masked OCR paths and are score-gated. Broad crop expansion and target-specific remapping are not included.

Closes #212
Refs #221

### CODE REVIEW

File: src/measure_numbering/mmr.py [hunk 1]
```diff
@@ -613,6 +614,82 @@ def _detect_number_with_evidence(self, image, system, x1, y1, x2, y2, prob, w_im
...
+                    _, best_score, best_dbg = max(
+                        [r for r in fallback_results if r[0] == current_num],
+                        key=lambda x: x[1],
+                    )
+                    if best_dbg:
+                        variant_debug = f"{best_dbg},variant=left_wide_unmasked_fallback_standard:0"
+                    else:
+                        variant_debug = "variant=left_wide_unmasked_fallback_standard:0"
+                    variant_results.append((current_num, best_score, variant_debug))
+
```
[the remark this answers is not in the document]
Comment (M763468) [reply src/measure_numbering/mmr.py]: 対応しました。

- duplicated fallback blocks を config loop に統合しました。
- masked path / unmasked fallback path の双方で empty crop guard を追加しました。
- empty crop が preprocessing に渡らないことを確認する regression test を追加しました。
- `make lint` は通過済みです。

fallback の順序と score gate は維持しています。

**Comment by M763468** [comment]
P2 review follow-up was evaluated and adopted in the latest push (`81592f8`).

Scope of the change:

- `unmasked_fallback_standard` now uses `self.rescue_threshold` as its minimum probability gate, so the current-crop unmasked fallback can participate in the existing rescue path.
- `left_wide_unmasked_fallback_standard` remains gated by `self.threshold`, so the wider crop is not opened in the rescue band.
- Added targeted tests for both behaviors.

Validation reported locally:

```text
make lint
All checks passed!
370 files already formatted

pytest tests/test_issue212_mmr_unmasked_fallback.py tests/test_issue213_mmr_one_bar_veto.py
14 passed
```

Full 68-page eval remains unchanged from the prior accepted candidate:

```text
Total Pages:         68
Total Base Measures: 3325
Total Expected:      182
Total Detected:      179
Matched (TP):        173
Missed (FN):         3
Skip Mismatch:       6
Unexpected (FP):     0
Precision:           0.9665
Recall:              0.9505
F1-Score:            0.9584
```

#213 veto remains preserved:

```text
[VETO] P33 S0 M1: Prob=0.54 -> OCR=11 (score=-27.4, one_bar_evidence=2)
```

**Comment by M763468** [comment]
Note: I confirmed the latest PR head is `81592f8` and the PR remains mergeable. The P2 change is visible in the diff: `unmasked_fallback_standard` is gated at `self.rescue_threshold`, while `left_wide_unmasked_fallback_standard` remains gated at `self.threshold`. The targeted tests now include the rescue-band current fallback and the threshold-only left-wide fallback.

**Comment by M763468** [comment]
I attempted to reply directly to the original P2 inline thread after adopting the change, but the review-comment reply endpoint could not resolve the parent comment id from the available thread payload. The validation and code confirmation are recorded above in the PR conversation comments.

**Comment by M763468** [comment]
Thread resolution note: the duplicated fallback / empty crop thread was already resolved. The active P2 issue has been addressed by implementation and validation in the latest head; if the inline thread still appears unresolved in the UI, it should be safe to mark it resolved based on the recorded PR validation.
